### PR TITLE
P2 updates

### PR DESCRIPTION
**Pull request by jerriep**
Updates to rules for P2

### CODE REVIEW

File: rules/creates-box-access-token.md [hunk 1]
```diff
@@ -40,15 +42,17 @@ function(user, context, callback) {
...
-      if(e) return callback(e,user,context);
-      user.persistent.box_id = u.id; //Save property in users' metadata
-      getBoxAccessToken(boxClientId, boxClientSecret, 
-                        u.id, "user", privateKey, function(e,token){
-        if(e) return callback(e,user,context);
-        user.box_access_token = token;
+      if(e) {
+        return callback(e,user,context);
+      }
+      user.app_metadata.box_id = u.id; //Save property in users' metadata
```
Comment (thameera) [inline rules/creates-box-access-token.md]: Should `user.app_metadata.box_id` be persisted here?

File: rules/facebook-custom-picture.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ function (user, context, callback) {
     // See: https://developers.facebook.com/docs/graph-api/reference/user/picture/ for more 
     // sizes and types of images that can be returned
     var pictureType = 'large';
-    user.picture = 'https://graph.facebook.com/v2.5/' + fbIdentity.user_id + '/picture?type=' + pictureType;
+    context.idToken.picture = 'https://graph.facebook.com/v2.5/' + fbIdentity.user_id + '/picture?type=' + pictureType;
```
Comment (thameera) [inline rules/facebook-custom-picture.md]: This should be namespaced with https://example.com/?

File: rules/pusher.md [hunk 3]
```diff
@@ -9,15 +9,15 @@ categories:
...
+```js
 function (user, context, callback) {
 
   var pusherKey='YOUR PUSHER KEY';
   var pusherSecret = '{YOUR PUSHER SECRET}';
  
   if( context.request.query.channel && context.request.query.socket_id)
   {
-    user.pusherAuth = pusherKey + ":" + sign(pusherSecret, context.request.query.channel, context.request.query.socket_id);
+    user['https://example.com/pusherAuth'] = pusherKey + ":" + sign(pusherSecret, context.request.query.channel, context.request.query.socket_id);
```
Comment (thameera) [inline rules/pusher.md]: s/user/context.idToken/ ?

File: rules/encrypt-sensitive-data.md [hunk 4]
```diff
@@ -10,12 +10,10 @@ This rule will set a sensitive value in the app_metadata and encrypt it (see the
 
 ```js
 function (user, context, callback) {
-  user.app_metadata = user.app_metadata || { };
-  user.app_metadata.private_data = encrypt({
+  user.idToken['https://example.com/private_data'] = encrypt({
```
Comment (thameera) [inline rules/encrypt-sensitive-data.md]: s/user/context/ ?

File: rules/zendesk-sso-jwt.md [hunk 5]
```diff
@@ -42,7 +42,7 @@ function (user, context, callback) {
   };
 
   var zendesk_token = jwt.sign(payload, configuration.ZENDESK_JWT_SECRET);
-  user.zendesk_jwt_url = 'https://' + ZENDESK_SUBDOMAIN + '.zendesk.com/access/jwt?jwt=' + zendesk_token;
+  user['https://example.com/zendesk_jwt_url'] = 'https://' + ZENDESK_SUBDOMAIN + '.zendesk.com/access/jwt?jwt=' + zendesk_token;
```
Comment (thameera) [inline rules/zendesk-sso-jwt.md]: s/user/context.idToken/ ?

File: rules/creates-box-access-token.md [hunk 6]
```diff
@@ -40,15 +42,17 @@ function(user, context, callback) {
...
+      getBoxAccessToken(boxClientId, boxClientSecret, u.id, "user", privateKey, function(e,token){
+        if(e) {
+          return callback(e,user,context);
+        }
+        user.idToken['https://example.com/box_access_token'] = token;
```
Comment (thameera) [inline rules/creates-box-access-token.md]: s/user/context/ ?

File: rules/creates-box-access-token.md (hunk 1 above)
Comment (jerriep) [inline rules/creates-box-access-token.md]: I believe so yes. There is a check on Line 28 to see whether user has already has been provisioned in Box. It looks at `user.app_metadata.box_id` to see whether this has been done

**Pull request merged**